src/Vizkit3dWorld.cpp

```diff
@@ -66,10 +68,8 @@ Vizkit3dWorld::Vizkit3dWorld(std::string path,
 Vizkit3dWorld::~Vizkit3dWorld()
 {
     app->closeAllWindows();
-    app->quit();
 
     delete widget;
-    delete app;
```

Comment:
You are leaking the QApplication instance

Reply:
@g-arjones, when we try to delete the app, the program crashes. Do you know an alternative way?

Reply:
Even after you fixed the issue with the scope of argc/argv? The alternative (and best option in my opnion) would be to not use a pointer here. You are instanciating it in the constructor and deleting in the destructor so this could very well be an attribute of the class

Reply:
> You are instantiating it in the constructor and deleting in the destructor so this could very well be an attribute of the class

Do you mean a static attribute? Something like is shown below
```c++

// header
class Vizkit3dWorld {

    QApplication app;
};

// source code
Vizkit3dWorld::Vizkit3dWorld() : app(argc, argv) 
{
}
```

We can try this.

Reply:
It doesn't have to be static but that's ok..